### PR TITLE
fix(ingest): fix lineage after dbt metadata ingestion when tables name and identifier differ

### PR DESCRIPTION
**Pull request by remisalmon**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

This PR fixes the lineage (upstream/downstream urns) generated after dbt metadata ingestion when tables tables name and identifier differ in the dbt manifest.json file:
- if `load_catalog` is False and a table (node) has an identifier field, use this instead of the table name to link to the actual source table in the urn
- if `load_catalog` is True, nothing is changed

See https://github.com/linkedin/datahub/pull/2483#issuecomment-840724995 for more context, and https://docs.getdbt.com/reference/resource-properties/identifier for the dbt doc on `identifier` vs `name`.

### CODE REVIEW

**Review by hsheth2 — APPROVED** [review]
LGTM, but it’d to be good to have @luck02 also take a look before merging

File: metadata-ingestion/src/datahub/ingestion/source/dbt.py [hunk 1]
```diff
@@ -98,17 +98,20 @@ def extract_dbt_entities(
         dbtNode = DBTNode()
 
         dbtNode.dbt_name = key
-        dbtNode.node_type = node["resource_type"]
-        dbtNode.name = node["name"]
         dbtNode.database = node["database"]
         dbtNode.schema = node["schema"]
         dbtNode.dbt_file_path = node["original_file_path"]
+        dbtNode.node_type = node["resource_type"]
+        if "identifier" in node and load_catalog is False:
```
Comment (luck02) [inline metadata-ingestion/src/datahub/ingestion/source/dbt.py:105]: Minor concern, there's no tests on this component and the integration test won't cover this.  

IMO I'd prefer to add tests before adding conditional logic but I think this is fine as there's no unit tests ATM.

**Review by luck02 — APPROVED** [review]
LGTM

**Review by shirshanka — APPROVED** [review]
LGTM!

**Comment by hsheth2** [comment]
@remisalmon could you run `./gradlew :metadata-ingestion:lintFix`?

**Comment by remisalmon** [comment]
@hsheth2 done.

**Review by shirshanka — APPROVED** [review]

**Pull request merged**